lib/mix/lib/mix/dep/fetcher.ex

```diff
@@ -64,12 +64,9 @@ defmodule Mix.Dep.Fetcher do
             scm.checkout(opts)
           end
 
-        if new do
```

Comment:
Removed this `if` since all SCMs should always return the lock.

Reply:
Path SCMs do not return a lock if there was no lock before. So I would add it back.